src/audio_processing.rs

```diff
@@ -0,0 +1,111 @@
+use dasp::Sample;
+use dasp_signal::Signal;
+use dasp_window::Window;
+use rustfft::{num_complex::Complex, num_traits::ToPrimitive};
+use std::sync::Arc;
+
+pub struct FftResult {
+    pub raw_bins: Vec<Complex<f64>>,
+    sample_rate: usize,
+    fft_size: usize,
+}
+
+impl FftResult {
+    pub fn new(raw_bins: Vec<Complex<f64>>) -> Self {
+        Self {
+            raw_bins,
+            sample_rate: 48000,
+            fft_size: 1024,
+        }
+    }
+
+    pub fn get_frequency_bin(&self, frequency: usize) -> Option<f64> {
+        let bin_size = (self.sample_rate / 2) / (self.raw_bins.len() / 2);
+        let bin = self.raw_bins.get(frequency / bin_size)?;
+        Some(bin.norm_sqr() / self.fft_size.to_f64().unwrap_or(1.0))
```

Comment:
why norm_sqr

Reply:
I see the Complex result as a 2d vector, and we use the length of that vector as "amplitude". So we use `sqrt(re*re + im*im)`

Reply:
@samichoulo911 ouais mais ce truc la etait pas deja fait a l'avance? Genre on pourrait le mettre direct dans le code de fft